EESSI-extend-easybuild.eb

```diff
@@ -196,10 +201,30 @@ elseif (project_modulepath ~= nil) then
   -- configure MODULEPATH
   prepend_path("MODULEPATH", project_modulepath)
 end
+
 -- Make sure EasyBuild itself is loaded
+-- need to also handle the unload behaviour where the version is defined only before we unload
+easybuild_version = os.getenv("EBVERSIONEASYBUILD")
 if not ( isloaded("EasyBuild") ) then
     load(latest("EasyBuild"))
 end
+easybuild_version = os.getenv("EBVERSIONEASYBUILD") or easybuild_version
+eessi_version = os.getenv("EESSI_VERSION") or "2023.06"
+
+-- Set environment variables that are EasyBuild version specific
+if convertToCanonical(easybuild_version) > convertToCanonical("4") then
+  setenv ("EASYBUILD_STRICT_RPATH_SANITY_CHECK", "1")
+  setenv ("EASYBUILD_CUDA_SANITY_CHECK_ERROR_ON_FAILED_CHECKS", "1")
+  setenv ("EASYBUILD_ENV_FOR_SHEBANG", "1")
+  setenv ("EASYBUILD_FAIL_ON_MOD_FILES_GCCCORE", "1")
+  setenv ("EASYBUILD_LOCAL_VAR_NAMING_CHECK", "error")
+  -- Set environment variables that are EESSI version specific
+  if convertToCanonical(eessi_version) > convertToCanonical("2023.06") then
+    setenv ("EASYBUILD_PREFER_PYTHON_SEARCH_PATH", "EBPYTHONPREFIXES")
+    setenv ("EASYBUILD_MODULE_SEARCH_PATH_HEADERS", "include_paths")
+    setenv ("EASYBUILD_SEARCH_PATH_CPP_HEADERS", "include_paths")
```

Comment:
Somewhat similar, why don't we just stick to the default (using `$CPATH`) here, is the case strong enough to go non-standard here by using `$C_INCLUDE_PATH` & co instead of `$CPATH`?

Reply:
The issue and resolution is in https://github.com/easybuilders/easybuild-framework/issues/3331#issuecomment-2694517824 , but that doesn't really help me to be any clearer on what we should do. 

@Flamefire You were involved in that discussion, what do you think? Is it enough to only change `EASYBUILD_MODULE_SEARCH_PATH_HEADERS`?

Reply:
I have seen issues fixed with the new `EASYBUILD_MODULE_SEARCH_PATH_HEADERS` but am not aware of the same for `EASYBUILD_SEARCH_PATH_CPP_HEADERS` but it seems sensible and I don't expect issues

I'd say this one is less problematic. If stuff really does break we can unset/change `EASYBUILD_SEARCH_PATH_CPP_HEADERS` because it isn't ingrained in module files except if the build did pick up wrong includes but I don't see how

Reply:
Yeah, that argument works in both directions

Reply:
I don't think it completely does go both ways. The problem is that `CPATH` is set in module files and that takes precedence, we can always increase precedence _in our builds_ by setting `CPATH` in a build (to the value of `C_INCLUDE_PATH`) if we need to. We could indeed also handle the reverse.

The issue is more on the user side, you are asking them to diagnose and do the same thing if _they_ run into the `CPATH` problem, whereas they won't see it in the other case.

Reply:
(see https://github.com/easybuilders/easybuild-framework/pull/4655/files)

Reply:
Ah, so @Flamefire 's earlier comment that this is more easily revertable

> If stuff really does break we can unset/change EASYBUILD_SEARCH_PATH_CPP_HEADERS because it isn't ingrained in module files except if the build did pick up wrong includes but I don't see how

applies to `EASYBUILD_SEARCH_PATH_CPP_HEADERS`, but changing `EASYBUILD_MODULE_SEARCH_PATH_HEADERS` is not (as easily) revertible, right?

Reply:
Same here, worst case we can actually back out later by patching or re-generating the module files, but it likely won't come to that.

To let's move forward by configuring EasyBuild so it doesn't fiddle with `$CPATH`